runtime/beta/auth/auth.go

```diff
@@ -60,7 +61,23 @@ func (mgr *Manager) Data() interface{} {
 // the empty string then data may not be nil. If these requirements are not met,
 // API calls made with these options will not be made and will immediately return
 // a client-side error.
+//
+// Note: this function will panic if the type of the data parameter is not
+// the same type returned by the auth handler. If the auth handler does not
+// return any data, then this value must be nil.
 func WithContext(ctx context.Context, uid UID, data interface{}) context.Context {
+	// If we're being passed a uid and data, make sure the data is valid.
+	if uid != "" || data != nil {
```

Comment:
Revert and preserve existing semantics (return a client-side error on the request)